pkg/pgmodel/lreader/labels_reader.go

```diff
@@ -78,13 +78,12 @@ func (lr *labelsReader) LabelValues(labelName string) ([]string, error) {
 			log.Debug("msg", "no tenants found for LabelValues()")
 			return []string{}, nil
 		}
-		existsClause, args := []string{}, []interface{}{}
+		existsClause, args := []string{}, []interface{}{labelName}
 		for i, tenantName := range validTenants {
-			existsClause = append(existsClause, fmt.Sprintf(tenantExists, i+1))
+			existsClause = append(existsClause, fmt.Sprintf(tenantExists, i+2)) // i+2 since $1 is reserved for labelName.
 			args = append(args, tenantName)
 		}
-		args = append(args, labelName)
-		labelValuesQuery := fmt.Sprintf(getLabelValuesForTenant, strings.Join(existsClause, " OR "), fmt.Sprint("$", len(args)))
+		labelValuesQuery := fmt.Sprintf(getLabelValuesForTenant, strings.Join(existsClause, " OR "))
```

Comment:
```suggestion
		labelValuesQuery := fmt.Sprintf(getLabelValuesForTenant, len(args), strings.Join(existsClause, " OR "))
```

Reply:
This will be wrong, as here, it will apply `$3` which is the size of args, but the requirement is for `$1`.